### ISSUE TITLE
cpython: tkinter missing on aarch64-darwin

### ISSUE THREAD

**Issue by angustrau** [NixOS/nixpkgs#133450]
### Describe the bug
The tkinter library is missing from `python3Full` on Apple Silicon.

### Steps To Reproduce

```
> nix-build . -A python3Full
/nix/store/i5h41g0iyi2z6pm5j0zzvsd5x3irh71f-python3-3.9.6

> ./result/bin/python3
Python 3.9.6 (default, Jul 26 2021, 23:27:45) 
[Clang 11.1.0 ] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import tkinter
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/angustrau/Documents/git/nixpkgs/result/lib/python3.9/tkinter/__init__.py", line 37, in <module>
    import _tkinter # If this fails your Python may not be configured for Tk
ModuleNotFoundError: No module named '_tkinter'
```

This was tested on master at `e221bcae106168e7e679767ac944f6adc961a933`.

### Expected behavior
This is the expected output, as recorded on an Intel mac:
```
> nix-build . -A python3Full
these paths will be fetched (12.25 MiB download, 55.34 MiB unpacked):
  /nix/store/b6xhm410fm5f25l4d7rwfq3156f47c72-tix-8.4.3
  /nix/store/k7a135szn4jxc1rjq0lsja22sl8zhnqx-python3-3.9.6
  /nix/store/x18zjwb2d8jga5qcnzl74x1isyy9h640-tcl-8.6.11
copying path '/nix/store/x18zjwb2d8jga5qcnzl74x1isyy9h640-tcl-8.6.11' from 'https://cache.nixos.org'...
copying path '/nix/store/b6xhm410fm5f25l4d7rwfq3156f47c72-tix-8.4.3' from 'https://cache.nixos.org'...
copying path '/nix/store/k7a135szn4jxc1rjq0lsja22sl8zhnqx-python3-3.9.6' from 'https://cache.nixos.org'...
/nix/store/k7a135szn4jxc1rjq0lsja22sl8zhnqx-python3-3.9.6
> ./result/bin/python3      
Python 3.9.6 (default, Jul 26 2021, 12:01:57) 
[Clang 7.1.0 (tags/RELEASE_710/final)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import tkinter
```

### Additional context
The `python3Packages.tkinter` package and all dependants are unavailable on M1 because of this bug.

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"aarch64-darwin"`
 - host os: `Darwin 20.6.0, macOS 11.5`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.4pre20210707_02dd6bb`
 - channels(angustrau): `""`
 - channels(root): `"nixpkgs-21.11pre303315.16105403bdd"`
 - nixpkgs: `/etc/nixpkgs`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
- python39Packages.tkinter
# a list of nixos modules affected by the problem
module:
```

**Comment by angustrau** [comment]
I've submitted a patch at #133750

**Comment by angustrau** [comment]
Closed by #133847
